rust/src/krb/krb5.rs

```diff
@@ -0,0 +1,560 @@
+/* Copyright (C) 2017-2018 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+// written by Pierre Chifflier  <chifflier@wzdftpd.net>
+
+use libc;
+use std;
+use std::ffi::{CStr,CString};
+use nom::{IResult,be_u32};
+use der_parser::der_read_element_header;
+use kerberos_parser::krb5_parser;
+use kerberos_parser::krb5::{EncryptionType,ErrorCode,MessageType,PrincipalName,Realm};
+use applayer;
+use core;
+use core::{AppProto,Flow,ALPROTO_FAILED,ALPROTO_UNKNOWN,STREAM_TOCLIENT,STREAM_TOSERVER};
+use conf::conf_get_bool;
+use parser::*;
+
+use log::*;
+
+#[repr(u32)]
+pub enum KRB5Event {
+    MalformedData = 0,
+    WeakCrypto,
+}
+
+pub struct KRB5State {
+    pub req_id: u8,
+
+    /// Configuration option to warn user on weak crypto
+    pub log_weak_crypto: bool,
+
+    /// List of transactions for this session
+    transactions: Vec<KRB5Transaction>,
+
+    /// tx counter for assigning incrementing id's to tx's
+    tx_id: u64,
+}
+
+pub struct KRB5Transaction {
+    /// The message type: AS-REQ, AS-REP, etc.
+    pub msg_type: MessageType,
+
+    /// The client PrincipalName, if present
+    pub cname: Option<PrincipalName>,
+    /// The server Realm, if present
+    pub realm: Option<Realm>,
+    /// The server PrincipalName, if present
+    pub sname: Option<PrincipalName>,
+
+    /// Encryption used (only in AS-REP and TGS-REP)
+    pub etype: Option<EncryptionType>,
+
+    /// Error code, if request has failed
+    pub error_code: Option<ErrorCode>,
+
+    /// The internal transaction id
+    id: u64,
+
+    /// The detection engine state, if present
+    de_state: Option<*mut core::DetectEngineState>,
+
+    /// The events associated with this transaction
+    events: *mut core::AppLayerDecoderEvents,
+
+    logged: applayer::LoggerFlags,
+}
+
+pub fn to_hex_string(bytes: &[u8]) -> String {
+    let mut s = String::new();
+    for &b in bytes {
+        s.push_str(&format!("{:02X}", b));
+    }
+    s
+}
+
+impl KRB5State {
+    pub fn new() -> KRB5State {
+        let l = conf_get_bool("app-layer.protocols.krb5.warn-weak-crypto");
```

Comment:
this is done at packet runtime, right? The conf api isn't really designed for that. Can we do this at init somehow?

Reply:
Sure, done in a commit added to the list (ec9f31d). If you prefer me to squash the commits just tell me